presto-main/src/main/java/io/prestosql/spiller/FileSingleStreamSpillerFactory.java

```diff
@@ -183,4 +189,21 @@ private boolean hasEnoughDiskSpace(Path path)
             throw new PrestoException(OUT_OF_SPILL_SPACE, "Cannot determine free space for spill", e);
         }
     }
+
+    private boolean isAccessible(Path path)
+    {
+        return isReadable(path) && isWritable(path) && isExecutable(path);
+    }
+
+    // Test whether a dir is working correctly by actually creating a random file in it
+    private boolean isSeeminglyHealthy(Path path)
+    {
+        try {
+            Path healthTemp = createTempFile(path, "spill", "healthcheck");
+            return deleteIfExists(healthTemp);
+        } catch (IOException e) {
+            log.warn(e, "Health check failed for spill %s", path);
```

Comment:
in case it does not work, should not you mark that this spill path is not usable and we should not consider this path in future.

Reply:
I explained a disk failure handling automation under the linked issue under which a bad disk can be brought back to rotation without restarting the service using the disk, i.e. the disk will not remain 'bad' forever. In that scenario, there is case for re-checking a previously discarded spill path. 

But I agree it can be optimized instead of checking each time. However, I am not too familiar with hotness of this code path, assuming your concern is related to performance.